operator-framework-core/src/test/java/io/javaoperatorsdk/operator/LeaderElectionManagerTest.java

```diff
@@ -3,28 +3,37 @@
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Optional;
 
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.io.TempDir;
 
+import io.fabric8.kubernetes.api.model.authorization.v1.ResourceRule;
+import io.fabric8.kubernetes.api.model.authorization.v1.SelfSubjectRulesReview;
+import io.fabric8.kubernetes.api.model.authorization.v1.SubjectRulesReviewStatus;
 import io.fabric8.kubernetes.api.model.coordination.v1.Lease;
 import io.fabric8.kubernetes.client.Config;
 import io.javaoperatorsdk.operator.api.config.ConfigurationService;
 import io.javaoperatorsdk.operator.api.config.LeaderElectionConfiguration;
 
 import static io.fabric8.kubernetes.client.Config.KUBERNETES_AUTH_TRYKUBECONFIG_SYSTEM_PROPERTY;
 import static io.fabric8.kubernetes.client.Config.KUBERNETES_NAMESPACE_FILE;
+import static io.javaoperatorsdk.operator.LeaderElectionManager.COORDINATION_GROUP;
+import static io.javaoperatorsdk.operator.LeaderElectionManager.LEASES_RESOURCE;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
 class LeaderElectionManagerTest {
 
-  private LeaderElectionManager leaderElectionManager() {
+  private LeaderElectionManager leaderElectionManager(Optional<Object> selfSubjectReview) {
```

Comment:
It would be better to not use Optional here but pass the object directly and use `Optional.ofNullable` when resolving the kubernetes client instead.